### PR TITLE
Don't render an ad label if the ad is hidden

### PR DESCRIPTION
**Pull request by arelra**
## What does this change?

The ad label text `Advertisement` was appearing for ads marked as 'excluded' in Google Ad Manager.

Excluded ads are served as 2x2 creatives which have the class `u-h` added by [render-advert.js](https://github.com/guardian/frontend/blob/ravi/dont-show-ad-label-for-ad-exclusions/static/src/javascripts/projects/commercial/modules/dfp/render-advert.js#L156)

This change checks if the ad slot has a `u-h` class in which case the ad label is not rendered by [render-advert-label.ts](https://github.com/guardian/frontend/blob/ravi/dont-show-ad-label-for-ad-exclusions/static/src/javascripts/projects/commercial/modules/dfp/render-advert-label.ts#L15)

## Does this change need to be reproduced in dotcom-rendering ?

- [X] No
- [ ] Yes (please indicate your plans for DCR Implementation)

## Screenshots

Before:

1)

https://www.theguardian.com/gnmeducationcentre

Screenshot 2021-05-11 at 17 12 44

2)

https://www.theguardian.com/commentisfree/2021/may/07/my-kids-wont-even-pick-up-a-book-and-it-is-definitely-all-my-fault

Screenshot 2021-05-11 at 17 10 40

After:

1)

Screenshot 2021-05-11 at 17 08 33

2)

Screenshot 2021-05-11 at 17 09 22

### Tested

- [X] Locally
- [ ] On CODE (optional)

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -1900,6 +1900,11 @@
   resolved "https://registry.yarnpkg.com/@guardian/eslint-config/-/eslint-config-0.5.0.tgz#ece14ec90b372932c7bf01992fc6d8ce58f7f26e"
   integrity sha512-THJn0wg7i9YjSVM9ofobsJgS+WH9ybrZH1+0CXLAebp0T53RGtgZUkek82mDg1h3DidsM+f7HSZqoQyrCbPzPw==
 
+"@guardian/libs@^1.7.1":
+  version "1.8.1"
+  resolved "https://registry.yarnpkg.com/@guardian/libs/-/libs-1.8.1.tgz#22bccd66be7c5bf70cd5bbb1bf7700d94610ccb7"
+  integrity sha512-Q/JPLhNeYa5XhDPJhw/1+fbZmszopvovWW8I7dGEuFPGWeNUmXAbW8J3kzuonDu7l9/tuFZg8jljbKHZlrFx0A==
+
 "@guardian/libs@^1.8.0":
```
Comment (mxdvl) [inline yarn.lock:1903]: How weird is it that these two versions don't resolve to the same dependency?

**Review by mxdvl — APPROVED** [review]
It's good to have the tests capture this case specifically.

File: yarn.lock (hunk 1 above)
Comment (arelra) [reply yarn.lock:1903]: That is strange. I would have thought the carat would make them align. Could a transitive dependency be causing this somehow?

Comment (mxdvl) [reply yarn.lock:1903]: Did you re-run `make install`?

Comment (arelra) [reply yarn.lock:1903]: Yep I reran and no different

**Comment by jfsoul** [comment]
Hey, coincidentally I was curious about this yesterday and had a quick look at some of the related code. I spotted that some of the ad slots are rendered with `data-label="false"` on frontend, but no such attribute on DCR. Haven't proved it yet, but might be an alternative solution to add that attribute?

**Comment by arelra** [comment]
> Edit: confirmed adding `data-label="false"` to the appropriate ad slots in AdSlot.tsx achieves the same result. Interesting though that both methods still leave a grey bar, which is presumably unintentional (frontend doesn't have this AFAICT)

Thanks for the comment.

Apologies for the length of this reply but there are some complexities here and I thought it would be useful for the wider team.

### TLDR

So although we want the ad slot to disappear we have two separate (but related) functions being performed:

1. "hiding" the ad label: `data-label=false` does this
2. "collapsing" the ad slot so it is not visible: `u-h` does this in Frontend and we should have the equivalent in DCR

And longer term we should come up with a better way of excluding ads.

### Background

The ad ops team have "exclusion" line items that are meant to _not_ show ads on specific pages and ad slots.

They do this by a (not very nice) hack of setting the ad creative to be 2x2 pixels image [details](https://sites.google.com/guardian.co.uk/digitalops-ad-issues-guide/actioning-a-pause-request/blocking-tag-pages/implementing-tag-blocks)

The commercial code then detects this size of ad and sets the `u-h` class on the ad slot.

The `u-h` [css class](https://github.com/guardian/frontend/blob/5b970cd7308175cfc1bcae2d4fb8c06ee13c5fa0/static/src/stylesheets/_mixins.scss#L73) (in Frontend) sets the ad slot to be 1x1 and positions it off screen.

(ps given the comment on u-h _"Hide content visually, still readable by screen readers"_ I think this is potential misuse of this class)

### Bug

#### top-above-nav

For `top-above-nav` slots I recently changed the way labels were created to remove the layout shift when they are added to the ad slot.

That change moved the label outside of the ad slot to be a sibling rather than being dynamically inserted.

This meant when we have excluded ads, the ad labels were no longer being made invisible by `u-h` as they were no longer a child of the ad slot.

This change fixes that.

#### other slots on articles rendered by DCR

The labels for other slots remain a child of the ad slot so if `u-h` is applied the slot and label would entirely disappear (on Frontend).

As you mentioned in the ticket on DCR, although the ad is not shown (because it is an empty 2x2) the ad label is still there.

I believe this because DCR  does not provide the equivalent css for `u-h`.

As you said adding `data-class=false` attribute to the ad slot will not display the ad label when the ad is rendered or refreshed.

However we are still left with the ad slot itself, as it does not get "collapsed".

### Short term fix

**Labels:** Pushing this change will fix the appearance of ad labels in all cases as the ad label render code will detect a `u-h` class on the ad slot and `display: none` the label.

**Collapsing:** We will need to add a change to DCR to collapse the slot, preferably not in the way 'u-h' works and I assume something more declarative and an underlying `display: none` would be preferable.

### Longer term

As you can see we need to revisit the way we collapse ads and render labels.

Ideally I'd like to set a class / attribute in GAM which could be passed through to signal not to render _anything_ at all rather than rendering the ad iframe and then hiding. This will require a wider discussion with the ad ops team.

We'll start the ball rolling on this.

**Comment by mxdvl** [comment]
Indeed, relying on adding a data attribute like `data-collaped="true"` is probably a more appropriate way than using this `u-h` class, which is still visible to screen readers.
